app/controllers/saml_idp_controller.rb

```diff
@@ -56,7 +56,7 @@ def render_template_for(message, action_url, type)
   end
 
   def delete_branded_experience
-    session.delete(:sp) if session[:sp]
-    session.delete(:user_return_to) if session[:user_return_to]
+    session.delete(:sp)
+    session.delete('user_return_to')
```

Comment:
I don't think so. I personally prefer symbols. If this string is bothering you, perhaps we shouldn't even call this session key directly and use Devise's `stored_location_for` since it has some logic that might be necessary and that we are ignoring, namely the scenario where `is_navigational_format?` return `false`, in which case, `session['user_return_to']` should be called instead of `session.delete('user_return_to')`. I don't understand navigational formats to know when that might happen.